modules/claims_api/app/models/claims_api/veteran.rb

```diff
@@ -65,8 +67,8 @@ def authn_context
       'authn'
     end
 
-    def set_edipi(edipi_header)
-      self.edipi = edipi_header.presence || mvi.profile&.edipi
+    def edipi(edipi_header = nil)
```

Comment:
We shouldn't have edipi as an attribute and a method. We should either use the old set_edipi method to set the attribute or remove it as a model attribute.

Reply:
I think the reason it was failing before is that it wasn't ever actually setting the self.edipi which is why it kept coming back nil even with your changes. Referencing the @edipi ensured that the attribute value was filled in and making it a method ensured reference. Making it a both an attribute and a method means it'll always catch it no matter where it's referenced and from all of my testing works in ensuring that the EDIPI is never nil.

Reply:
Can you add a spec that shows that? I am not following how the method doesnt cancel out the attribute.

Reply:
so I've tried it every other way and we seem to need both the attributes_attr and the similar named method for assignment. Hear me out here on the why. When you called set_edipi from the self.from_headers method, loa was never set yet so your fall back for mvi.profiles&.edipi would never actually get called because mvi would come back with nothing because loa 3 wasn't set yet. So by having a fall back method that checks the edipi_header.presence and if that still doesn't exist fall back to mvi.profile&.edipi ensures that even it doesn't load via the initial headers, if it's called at any point later it will reference the edipi in the mvi after loa is set. You need both the attributes set and the method as a fall back because you don't know at which stage the edipi method/variable will be called.